### DISCUSSION TITLE
Find if post has been published at least once

### DISCUSSION THREAD

**Discussion by pmatos** [11ty/eleventy#2623] [Q&A]
Hi,

Is there a way to know if a post has been published at least once?

This is to avoid the regeneration of some assets I have as filters. I am worried that if I always regenerate everything, I might regenerate old assets that will change old blog posts and I never notice it.

Has anyone faced this problem?

**Comment by btrem** [comment]
I don't understand the question. (So apparently I have *not* faced this problem.) :(

What do you mean by "published at least once"? Do you mean if an output file was generated by 11ty? Or if it was uploaded to a website in some manner? If the latter, then no. Sadly for you, I think the answer is still no if you meant the former. But could you explain a bit more what problem you have?

**Comment by pdehaan** [comment]
Yeah, I don't think it's possible/easy to know if a post has been published and prevent it from being published again.

If your main concern is to avoid [needlessly] regenerating assets, I'd probably just check in the processed assets into source control instead of processing them on each build via filters or shortcodes, etc.
There is also https://www.11ty.dev/docs/plugins/fetch/ which potentially let you cache the results of the filter for an arbitrary length of time (like a month or year or infinity). That might be better performance wise than regenerating assets via filters on each build, but wouldn't solve the concern of regenerating old assets unexpectedly.

Only potential solution I could think about was some overly complex system of keeping some SHA1 (or whatever) hash of assets and then finding ways to store that with each post to tell you if some asset has changed. Something like https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity. That way if an image changed, you might be able to detect that and throw some build error. But that feels like a lot of work.

**Comment by pmatos** [comment]
Thanks for @pdehaan and @btrem for attempting to answer my poorly formulated question.

The issue is really regarding assert regeneration.
I have a filter that given a markup file, runs some scripts to generate an svg that's automatically embedded in a post. My concern is that I might in the future change how this script works, and regeneration will generate a new svg for an old post. If for some reason this svg (for an old post) is now broken, I will never notice.

Therefore having a way to say, hey, if I regenerated this in the past and posted it online, then don't regenerate.

I guess my partial answer to this is that I should publish the generated assets into the repo, and ensure that each time I change the filter, none of the past svg's change.

Any other ideas that don't expect me to pay attention to these changes are welcome. :)
